app/controllers/IncomeController.scala

```diff
@@ -309,32 +311,34 @@ class IncomeController @Inject()(
       Redirect(confirmationCallback)
     }
 
-  def editPensionIncome(): Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
-    implicit val user: AuthedUser = request.taiUser
+  def editPensionIncome(empId: Int): Action[AnyContent] = (authenticate andThen validatePerson).async {
+    implicit request =>
+      implicit val user: AuthedUser = request.taiUser
 
-    journeyCacheService
-      .collectedJourneyValues(
-        Seq(UpdateIncome_PayToDateKey, UpdateIncome_IdKey, UpdateIncome_NameKey),
-        Seq(UpdateIncome_DateKey))
-      .getOrFail flatMap {
-      case (mandatorySeq, optionalSeq) =>
-        val date = Try(optionalSeq.head.map(date => LocalDate.parse(date))) match {
-          case Success(optDate) => optDate
-          case Failure(exception) =>
-            logger.warn(s"Unable to parse updateIncomeDateKey $exception")
-            None
-        }
+      journeyCacheService
+        .collectedJourneyValues(
+          Seq(UpdateIncome_PayToDateKey, UpdateIncome_IdKey, UpdateIncome_NameKey),
+          Seq(UpdateIncome_DateKey))
+        .getOrFail flatMap {
+        case (mandatorySeq, optionalSeq) =>
+          val date = Try(optionalSeq.head.map(date => LocalDate.parse(date))) match {
```

Comment:
no need for a Try or anything here... (Throwing an exception is expensive and useless in this case :D ) 
optionalSeq.headOption.map(date => LocalDate.parse(date))

Reply:
This will make date an option so will have extract this out, how do you want me to do thi?

Reply:
it's fine as it is